### DISCUSSION TITLE
Automated way of deleting unused translation json entries

### DISCUSSION THREAD

**Discussion by anaclumos** [facebook/docusaurus#10366] [General]
Is there an automated method for removing unused translation JSON entries?

**Comment by slorber** [comment]
We don't provide this as a first-class feature, but we have a way to do this internally for our own default translations:

`yarn workspace @docusaurus/theme-translations update`

![CleanShot 2024-08-05 at 10 16 49@2x](https://github.com/user-attachments/assets/8a4f56eb-1f95-40ea-9dbe-08a71874c789)

You can take inspiration from our code here:
https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-translations/update.mjs

Note that although we recommend using static strings for translation ids, you can also use dynamic onces in practice, using data coming from an external source for example. For this reason we can't reliably extract 100% of the actual translation ids of a codebase, and can't 100% reliably know for sure that a translation is not used. We might know a translation is not used statically, but it might be used dynamically, and if we report if, it can be a false positive report.

For example:

```js
<Translate id={`users.${username}`>Username</Translate>
```

We can't tell if `users.slorber` is used or not reliably.
